src/app/meshes/data/index.ts

```diff
@@ -0,0 +1,84 @@
+import { getDataplaneStatusCounts } from '@/app/data-planes/data'
+import { getServiceTypeCount } from '@/app/services/data'
+import type { PaginatedApiListResponse } from '@/types/api.d'
+import type {
+  Backend,
+  DataPlaneProxyStatus as DataplaneStatusCount,
+  MeshBackend,
+  Mesh as PartialMesh,
+  MeshInsight as PartialMeshInsight,
+  ServiceStatus as ServiceTypeCount,
+} from '@/types/index.d'
+
+export type Mesh = PartialMesh & {
+  mtlsBackend: Backend | undefined
+  metricsBackend: Backend | undefined
+}
+
+export type MeshInsight = PartialMeshInsight & {
+  dataplanes: Required<DataplaneStatusCount>
+  dataplanesByType: {
+    standard: Required<DataplaneStatusCount>
+    gateway: Required<DataplaneStatusCount>
+    gatewayBuiltin: Required<DataplaneStatusCount>
+    gatewayDelegated: Required<DataplaneStatusCount>
+  }
+  services: Required<ServiceTypeCount>
+  totalPolicyCount: number
+}
```

Comment:
I went slightly differently about this now because I also need this for nested structures (i.e. `dataplanesByType`) while still relying on the underlying partial type:

```ts
export type MeshInsight = PartialMeshInsight & {
  dataplanes: Required<PartialMeshInsight['dataplanes']>
  dataplanesByType: {
    standard: Required<PartialMeshInsight['dataplanesByType']['standard']>
    gateway: Required<PartialMeshInsight['dataplanesByType']['gateway']>
    gatewayBuiltin: Required<PartialMeshInsight['dataplanesByType']['gatewayBuiltin']>
    gatewayDelegated: Required<PartialMeshInsight['dataplanesByType']['gatewayDelegated']>
  }
  services: Required<PartialMeshInsight['services']>
  totalPolicyCount: number
}
```